metrics/report/report_dockerfile/collectd_scaling.R

```diff
@@ -28,8 +29,11 @@ iferrordata=c()		# Track interface errors data for nodes
 memstats=c()		# Statistics for memory usage
 cpustats=c()		# Statistics for cpu usage
 bootstats=c()		# Statistics for boot (launch) times
+bootmeanstats=c()	# Statistics for boot (launch) times mean bucket diffs
 inodestats=c()		# Statistics for inode usage
 
+bootdiffbuckets=10	# How many buckets do we evaluate the boot times in
```

Comment:
Depending on the amount of data, having the buckets fixed may hide a data distribution that can be useful to spot. I'm not sure if R have an automatic bucket recommendation, I'm sure that `matlab` and and `octave` does.

Reply:
Absolutely. It's also an issue if an 'interesting data point' happens to fall on a boundary between two slices. It is a very similar question to how much 'smoothing' we do to the data with the smoothed or rolling mean lines. Note, this bucketing (possibly better called slicing?) code is **not** a histogram - we are bucketing the linear (time based) data.
There is also the issue of if we get more buckets, they won't fit into the text table. The general idea of the page is to give a 'quick overview' to the questions such as:
- what is the cost per-container between launch `n` and `n+1`?
- is the launch time overhead fairly linear and constant?

If we take bucket number selection to the extremes then we have:
- at minimum, we would show one bucket. That is basically the same as using a single variable `lm()` (flat line), or the mean of the whole data set. That's fine if the data is linear, but I don't think our data is generally that linear
- at maximum, we would show one bucket per sample. That is the equivalent of plotting out the raw 'diff' data - see the graph I posted above - that data is pretty noisy.

I'm probably coming to the conclusion that buckets are not the ideal solution, and that some form of smoothing or rolling mean is better - visually it gives a much better representation of the 'average' boot diff time for each sample point, and not just for 10 individual sub-ranges. I'm leaning towards not calculating/drawing the buckets - but using a smoothed line, and taking 10 sample points off that line for the table data [iyswim](https://acronyms.thefreedictionary.com/IYSWIM).

OOI, I did have a look around for what R might offer in the 'auto bucketing' methods. Yes, there are some, and they tend to be used with the [histograms](https://stat.ethz.ch/R-manual/R-devel/library/graphics/html/hist.html). See the [nclass()](https://stat.ethz.ch/R-manual/R-devel/library/grDevices/html/nclass.html) functions for instance.
If we were going to auto-bucket then I think we'd use those. I may have a play with them, but I have a feeling at some point they would return a large number, and that would blow the text table width out of the page...